src/printer.js

```diff
@@ -2062,8 +2062,12 @@ function printFunctionParams(path, print, options) {
     return concat(["(", join(", ", printed), ")"]);
   }
 
+  const isFlowShorthandWithOneArg = fun.type === "FunctionTypeAnnotation" &&
```

Comment:
Ok, done. But I needed to create a similar one called `isTypeAnnotationAFunction` and also keep a checking for TypeAlias too 😄.